src/_C014.ino

```diff
@@ -0,0 +1,686 @@
+#endif
+
+// subscribe and publish shemes should not be changed by the user. This will probably break the homie convention. Do @ your own risk;)
+#define CPLUGIN_014_SUBSCRIBE       "homie/%sysname%/#" // "homie/%sysname%/+/+/#" causes problems (!ToDo)
+#define CPLUGIN_014_PUBLISH         "homie/%sysname%/%tskname%/%valname%"
+
+#define CPLUGIN_014_BASE_TOPIC      "homie/%sysname%/#"
+#define CPLUGIN_014_BASE_VALUE      "homie/%sysname%/%device%/%node%/%property%"
+#define CPLUGIN_014_INTERVAL        "90" // to prevent timeout !ToDo set by towest plugin interval
+#define CPLUGIN_014_SYSTEM_DEVICE   "SYSTEM" // name for system device Plugin for cmd and GIO values
+#define CPLUGIN_014_CMD_VALUE       "cmd" // name for command value
+#define CPLUGIN_014_GPIO_VALUE      "gpio" // name for gpio value i.e. "gpio1"
+#define CPLUGIN_014_CMD_VALUE_NAME  "Command" // human readabele name for command value
+
+byte msgCounter=0; // counter for sent Messages
+
+// !ToDo: should be integrated in bool CPlugin_014();
+bool CPlugin_014_Feedback(byte Function, struct EventStruct *event, String& str)
+{
+  MakeControllerSettings(ControllerSettings);
+  LoadControllerSettings(event->ControllerIndex, ControllerSettings);
+
+  if (str!="") {
+    String commandName = parseString(str, 1);
+    int errorCounter = 0;
+    if (commandName == F("gpio")) //!ToDo : As gpio is like any other plugin commands should be integrated below!
+    {
+      int port = parseString(str, 2).toInt();
+      int valueInt = parseString(str, 3).toInt();
+      String valueBool = "false";
+      if (valueInt==1) valueBool = "true";
+
+      String log = F("C014 : Feedback GPIO");
+      log += port;
+      log += F(" value:");
+      log += valueBool;
+      log += F(" (");
+      log += valueInt;
+      log += F(")");
+
+      String topic = ControllerSettings.Publish;
+      topic.replace(F("%sysname%"), Settings.Name);
+      topic.replace(F("%tskname%"), CPLUGIN_014_SYSTEM_DEVICE);
+      topic.replace(F("%valname%"), CPLUGIN_014_GPIO_VALUE + toString(port,0));
+
+      if (MQTTpublish(CPLUGIN_ID_014, topic.c_str(), valueBool.c_str(), errorCounter)) {addLog(LOG_LEVEL_INFO, log+" success!");}
+        else {addLog(LOG_LEVEL_ERROR, log+" ERROR!");  return false;}
+      return true;
+    } else
+    {
+      String topic = ControllerSettings.Publish;
+      topic.replace(F("%sysname%"), Settings.Name);
+      int deviceIndex = parseString(str, 2).toInt();
+      LoadTaskSettings(deviceIndex-1);
+      String deviceName = ExtraTaskSettings.TaskDeviceName;
+      topic.replace(F("%tskname%"), deviceName);
+      String valueName = ExtraTaskSettings.TaskDeviceValueNames[parseString(str, 3).toInt()-1];
+      topic.replace(F("%valname%"), valueName);
+
+      String log = F("C014 : Feedback ");
+      log += deviceName;
+      log += F(" var: ");
+      log += valueName;
+      log += F(" topic: ");
+      log += topic;
+      log += F(" value: ");
```

Comment:
Wrap in check for `logLevelActiveFor`

Reply:
Didn't notice that this exists, that's perfect, will do that for all logs. Will take a wile (there are many of them (perhaps too many).

Newbie question: What is the best way to do this? Do the changes locally and then commit?

Reply:
Yep, you should continue on the branch you made on your local computer.
When you make a commit and push it, it will be appended to this pull request.

A pull request is about merging a branch into another branch.
So adding commits to the branch you want to merge will be included in this PR. (as soon as you push a commit)

Reply:
So I hope I addressed all of your comments. I think it is time to commit the changes.
Let`s give it a try ;)